### PR TITLE
Make coreblocks have their team represented on sprite 2: Electric Boogaloo

### PR DESCRIPTION
**Pull request by Quezler**
This pull is just a complete & cleaned up version of #2166 <-- credits obviously go there ❤️ 

(doesn't contain repacked sprites since their diff looks a bit to weird compared to previous)

todo:
- render it correctly in the editor
- somehow avoid having sharded hardcoded
- etc

Screen Shot 2020-06-19 at 20 14 09

### CODE REVIEW

File: core/src/mindustry/world/blocks/storage/CoreBlock.java [hunk 1]
```diff
@@ -164,6 +168,18 @@ public void onProximityUpdate(){
...
+            if(team == Team.sharded){
+                Draw.rect(block.region, x, y, block.rotate ? rotdeg() : 0);
+            }else{
+                Draw.rect(baseRegion, x, y, block.rotate ? rotdeg() : 0);
+                Draw.color(team.color);
+                Draw.rect(topRegion, x, y, block.rotate ? rotdeg() : 0);
+                Draw.color();
+            }
+        }
+
```
Comment (Quezler) [inline core/src/mindustry/world/blocks/storage/CoreBlock.java:182]: not sure if you want to have `, block.rotate ? rotdeg() : 0` there or not ^ (modding support?)

File: core/src/mindustry/world/blocks/storage/CoreBlock.java [hunk 2]
```diff
@@ -181,6 +197,10 @@ public void drawSelect(){
             Draw.reset();
         }
 
+        @Override
+        public void drawTeam(){
+            //
```
Comment (Quezler) [inline core/src/mindustry/world/blocks/storage/CoreBlock.java:202]:
```suggestion
            //no
```

**Comment by ThePythonGuy3** [comment]
Thanks for making my PR better!

**Comment by Quezler** [comment]
You're welcome @ThePythonGuy3 ❤️

**Comment by Quezler** [comment]
for @Anuken's reference, this is how the sharded core would currently look if not excluded:

Screen Shot 2020-06-19 at 20 38 20

**Comment by Quezler** [comment]
What i'm noticing however is that just like some other pulls i worked on (like #1884 for example) there tend to be white/transparent/alignment/etc issues where the texture overlays meet, any idea how to best remedy those whenever they occur?

**Comment by ThePythonGuy3** [comment]
> for @Anuken's reference, this is how the sharded core would currently look if not excluded:
> 
> Screen Shot 2020-06-19 at 20 39 52
> 
> 

That's why I added a shared team sprite also

**Comment by ThePythonGuy3** [comment]
> What i'm noticing however is that just like some other pulls i worked on (like #1884 for example) there tend to be white/transparent/alignment/etc issues where the texture overlays meet, any idea how to best remedy those whenever they occur?

I ran inte the exact same problem, but as my PR didnt compile in the newer versions of v6, I did not add the solution to mine. But the solution is simple: you need to make the base be "fluent" with the colors. Like this:
 image

**Comment by Anuken** [comment]
> you need to make the base be "fluent" with the colors.

The reason colors "bleed" through is due to the antialiasing/filtering of the block underneath; the base edges under the "top" region blend into the nearby colors and make them lighter. That's why coloring the base sprite properly fixes it.

---

A true solution to the problem with shading could be to automatically generate sprites at pack-time with the correct palette and shading, then have the cores use a generated sprite based on team. Of course, only the important teams (crux, sharded, maybe derelict) would get sprites - other teams would use the fallback tinted sprite. 

This has several advantages:

- You'll only need the top sprite and the bottom sprite, no other copies or manual resprites need to be made
- Any block can be made to use a team color pattern - all you need is the top sprite
- Once you define a team palette, it can be used for every tinted team block - no need to remember the exact colors

**Comment by Quezler** [comment]
well i suppose rewriting the pack engine to do the above falls above my scope so I'll archive this 😗

**Comment by Anuken** [comment]
Final result:

image

**Comment by Quezler** [comment]
you've kept the team stripes in the bottom left corner? (graphics look nice btw) ❤️

**Comment by Anuken** [comment]
I *could* remove them, but they aren't much of a problem right now.

**Pull request closed without merging**